File: DSA_Level-2/Bit Manipulation/All Repeating Three Times Except One/AllRepeatingThreeTimesExceptOne_Solution.java
import java.io.*;
import java.util.*;

public class Main {

  public static void main(String[] args){
    Scanner scn = new Scanner(System.in);
    int n = scn.nextInt();
    int[] arr = new int[n];
    for(int i = 0 ; i < n; i++){
      arr[i] = scn.nextInt();
    }
    solution(arr);
  }

  public static void solution(int[] arr){
   int tn=Integer.MAX_VALUE,tn1=0,tn2=0;
   for(int i=0;i<arr.length;++i){
       int cm=arr[i]&tn;
       int cm1=arr[i]&tn1;
       int cm2=arr[i]&tn2;
       
       tn=tn&(~cm);
       tn1=tn1|cm;
       
       tn1=tn1&(~cm1);
       tn2=tn2|cm1;
       
       tn2=tn2&(~cm2);
       tn=tn|cm2;
   }
   
   System.out.println(tn1);
  }

}

File: DSA_Level-2/Bit Manipulation/All Repeating Three Times Except One/AllRepeatingThreeTimesExceptOne_Solution.js

function allRepeatingThrice(nums){
    
    let tn=Number.MAX_SAFE_INTEGER, tn1=0, tn2=0;

    for(let num of nums){
        
        cm=tn&num;
        cm1=tn1&num;
        cm2=tn2&num;

        tn=tn&~cm;
        tn1=tn1|cm;

        tn1=tn1&~cm1;
        tn2=tn2|cm1;

        tn2=tn2&~cm2;
        tn=tn|cm2;

    }

    console.log(tn1);
}

allRepeatingThrice([1,1,1,2]);